lib/services/role.go

```diff
@@ -2079,6 +2102,33 @@ func (set RoleSet) EnhancedRecordingSet() map[string]bool {
 	return m
 }
 
+// HostUsers host user information matching a server or nil if a
+// role disallows host user creation
+func (set RoleSet) HostUsers(s types.Server) *HostUsersInfo {
+	var groups []string
+	serverLabels := s.GetAllLabels()
+	for _, role := range set {
+		result, _, err := MatchLabels(role.GetNodeLabels(types.Allow), serverLabels)
+		if err != nil {
+			return nil
+		}
+		// skip nodes that dont have matching labels
+		if !result {
+			continue
+		}
+		createHostUser := role.GetOptions().CreateHostUser
+		// if any of the matching roles do not enable create host
+		// user, the user should not be allowed on
+		if createHostUser == nil || !createHostUser.Value {
+			return nil
+		}
+		groups = append(groups, role.GetHostGroups(types.Allow)...)
+	}
+	return &HostUsersInfo{
+		Groups: groups,
```

Comment:
I think it's technically possible to make it here while groups is still empty and it looks like that might cause problems, would it make sense to check for that and return `nil` instead? alternatively make this return an error instead of just `nil`

Reply:
It seemed reasonable to me, but TBH we don't really use nil checks like that often, so maybe returning a `HostUsersInfo` that also contains an `Allowed` bool field could be more explicative?

Reply:
I was misreading this, I think I got it now, returning a valid `HostUsersInfo` with empty groups means we need to create the user with no groups, and `nil` means we don't need to create a user? I don't think there's a problem in that case, but the `nil` check for that is a bit uncommon as Edoardo mentioned